### PR TITLE
Enable custom themes for buttons

### PR DESCRIPTION
**Pull request by vNakamura**
affects: `@crave/farmblocks-button`

ISSUES CLOSED: #252

### CODE REVIEW

File: packages/button/src/styledComponents/Button.js [hunk 1]
```diff
@@ -80,8 +80,8 @@ function paddingStyle(props) {
   `;
 }
 
-function neutralStyle(props) {
-  const { textColor, iconColor, textHoverColor, color } = colorTypes[NEUTRAL];
+function neutralStyle(props, theme) {
```
Comment (fczuardi) [inline packages/button/src/styledComponents/Button.js]: theme on a component child of ThemeProvider is already present in the props, I think we could keep just one argument here with a little tweak below on the place this function is called

File: packages/button/src/styledComponents/Button.js [hunk 2]
```diff
@@ -105,10 +105,11 @@ function neutralStyle(props) {
 }
 
 function typeStyle(props) {
+  const theme = Object.assign({}, colorTypes, props.theme);
   if (props.type === NEUTRAL) {
-    return neutralStyle(props);
+    return neutralStyle(props, theme);
```
Comment (fczuardi) [inline packages/button/src/styledComponents/Button.js]: here (as discussed offline):
```
    return neutralStyle({...props, theme});
```

**Comment by vNakamura** [comment]
@fczuardi  changes addressed

File: packages/button/src/styledComponents/Button.js [hunk 3]
```diff
@@ -105,10 +105,11 @@ function neutralStyle(props) {
 }
 
 function typeStyle(props) {
+  const theme = Object.assign({}, colorTypes, props.theme);
```
Comment (alcferreira) [inline packages/button/src/styledComponents/Button.js]: Why not `spread operator` as you used on line 110? :P

`const theme = { ...colorTypes, ...props.theme };`

Comment (vNakamura) [reply packages/button/src/styledComponents/Button.js]: Because `props.theme` can be `undefined`

Comment (alcferreira) [reply packages/button/src/styledComponents/Button.js]: What if you set default `props.theme` to `{}`?

I personally prefer to use the spread operators, but if you guys prefer the `Object.assign` no problem

Comment (vNakamura) [reply packages/button/src/styledComponents/Button.js]: Yes! I prefer spread operators, too.
It didn't come to me that styled components could have defaultProps. I'll change it to spread.

**Pull request merged**